### PR TITLE
Add wrappers for application mode

### LINKED ISSUE #2245: App mode library doesn't work with new python tables
**Description**

The following application mode code does not work with the new python library

```
import jpy
from deephaven import empty_table
from typing import Callable

ApplicationState = jpy.get_type("io.deephaven.appmode.ApplicationState")

#ApplicationState examples
def start(app: ApplicationState):
    table = empty_table(5)
    table = table.update(["Values = i"])

    app.setField("table", table)

def initialize(func: Callable[[ApplicationState], None]):
  app = jpy.get_type("io.deephaven.appmode.ApplicationContext").get()
  func(app)

initialize(start)
```

Error thrown
```
server_1      | Type: <class 'ValueError'>
server_1      | Value: cannot convert a Python 'Table' to a Java 'java.lang.Object'
```

Changing the setField call to `app.setField("table", table.j_object)` makes it work as expected

**Expected results**

The code above works

**Actual results**

The code above errors out

**Additional details and attachments**

**Versions**
 - Deephaven: edge
 - OS: osx
 - Browser: chrome
 - Docker: Docker version 20.10.8, build 3967b7d

### PR DESCRIPTION
**Pull request by jmao-denver**
Fixes #2245

### CODE REVIEW

File: pyintegration/deephaven/appmode.py [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+    def __getitem__(self, item):
+        item = str(item)
+        value = wrap_j_object(self.j_app_state.getField(item))
+        if not value:
+            raise KeyError(item)
+
+    def __setitem__(self, key, value):
+        key = str(key)
+        self.j_app_state.setField(key, unwrap(value))
```
Comment (devinrsmith) [inline pyintegration/deephaven/appmode.py:38]: Note: the item you are returning from get is not the same item you are placing via set. I see you *are* calling `.value()` in fields() - I'm not sure if we want to have JObjectWrapper Field, or return `.value()` from a get.

**Review by jakemulf — DISMISSED** [review]
From the use case that I was looking to replicate, this looks good, but I'd consider Devin's feedback in case there's more use cases that I may have missed

File: pyintegration/deephaven/appmode.py (hunk 1 above)
Comment (jmao-denver) [reply pyintegration/deephaven/appmode.py:38]: I think Python dict is a good fit, wrapping Field seems a bit overkill.

**Review by jakemulf — APPROVED** [review]

File: pyintegration/deephaven/appmode.py (hunk 1 above)
Comment (devinrsmith) [reply pyintegration/deephaven/appmode.py]: Sure - but I just want to triple check:

```python
app = get_app_state()
app['hello'] = 'world'
print(app['hello'])
```
will *not* print `world` b/c getField returns a Field...

**Review by devinrsmith — COMMENTED** [review]
See earlier comment.

File: pyintegration/deephaven/appmode.py (hunk 1 above)
Comment (jmao-denver) [reply pyintegration/deephaven/appmode.py]: Thanks for finding the bug!

File: pyintegration/test_helper/__init__.py [hunk 2]
```diff
@@ -61,7 +63,13 @@ def start_jvm():
             'workspace': os.path.realpath(workspace),
 
         }
-        jvm_options= {
+
+        if jvm_props:
+            jvm_properties.update(jvm_props)
+
+        print(jvm_properties)
```
Comment (devinrsmith) [inline pyintegration/test_helper/__init__.py:70]: Was this on purpose, or meant to be debugging only?

Comment (jmao-denver) [reply pyintegration/test_helper/__init__.py:70]: good catch again, I really should have taken a break and not been rushing things.

**Review by jakemulf — APPROVED** [review]

**Pull request merged**